mlir/lib/Conversion/MemRefToLLVM/MemRefToLLVM.cpp

```diff
@@ -1562,8 +1562,6 @@ struct AtomicRMWOpLowering : public LoadStoreOpLowering<memref::AtomicRMWOp> {
   LogicalResult
   matchAndRewrite(memref::AtomicRMWOp atomicOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    if (failed(match(atomicOp)))
```

Comment:
I actually dont understand this check..... This looks OK to me, but what was stopped by this change?

Reply:
It looks okay to me. I don't know who should be involved in the review even I looked at `git blame`. It looks like very very few people update the pattern and op...

Reply:
> I actually dont understand this check..... This looks OK to me, but what was stopped by this change?

It fails when the memref type has an offset because it checks that the memref type has an identity map. But the pattern can handle offsets, so it should be fine to remove the check

Reply:
It seems like this only works for strided memref types. Can you check that `getStridesAndOffset` on the source type returns `success`, i.e. return failure when that method fails. I looked into the implementation of `getStridedElementPtr` and it asserts if thats not the case, so letting that through would lead to an assert